patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -653,11 +653,23 @@
  			}
  			return result;
  		}
-@@ -7490,7 +_,7 @@
+@@ -7480,17 +_,9 @@
  			{
- 				Main.SetGraphicsProfile(GraphicsProfile.HiDef);
+ 				TexturePackSupport.Enabled = true;
  			}
+-			Main.Configuration.Get<bool>("Support4K", ref Main.Support4K);
+-			bool flag = Main.Support4K && base.GraphicsDevice.Adapter.IsProfileSupported(GraphicsProfile.HiDef);
+-			if (GraphicsAdapter.DefaultAdapter.CurrentDisplayMode.Width <= 1920 && GraphicsAdapter.DefaultAdapter.CurrentDisplayMode.Height <= 1200)
```

Comment:
Rather than delete all this vanilla code, I think we should just comment out the check for display resolution, these 3 lines. Support4K defaults to true, so this would allow users to continue to use the Support4K config flag as designed to limit the HiDef profile.